src/jaegertracing/Tracer.cpp

```diff
@@ -258,6 +260,22 @@ Tracer::make(const std::string& serviceName,
     }
 
     auto metrics = std::make_shared<metrics::Metrics>(statsFactory);
+
+    TextMapPropagator* textPropagator;
+    HTTPHeaderPropagator* httpHeaderPropagator;
+    if (config.propagationFormat() == propagation::Format::W3C) {
+        textPropagator =
+            new propagation::W3CTextMapPropagator(config.headers(), metrics);
```

Comment:
Are headers() needed for W3C? If different headers are used it ceases to be W3C format.

Reply:
They are needed for the jaeger-debug-id header.